File: SIMPLEAPI_Demo/SIMPLEAPI_Demo/Impresion/Core/Helpers/StringHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIMPLEAPI_Demo.Impresion.Core.Helpers
{
    public static class StringHelper
    {
        public static string Truncate(this string value, int maxLength)
        {
            if (string.IsNullOrEmpty(value)) return value;
            if (value.Length > maxLength)
                Console.WriteLine(string.Format("Truncate performed: {0} to {1}", value, value.Substring(0, maxLength)));
            return value.Length <= maxLength ? value : value.Substring(0, maxLength);
        }

        public static string CenterTo(this string value, int lenght)
        {
            value = value.Trim();

            if (string.IsNullOrEmpty(value)) return value;
            if (value.Length > lenght) return value;
            return value.PadLeft(value.Length + ((lenght - value.Length) / 2), ' ');
        }
    }
}

File: SIMPLEAPI_Demo/SIMPLEAPI_Demo_VB/Impresion/Core/Helpers/StringHelper.vb
Imports System.Runtime.CompilerServices

Module StringHelper
    <Extension()>
    Public Function Truncate(ByVal value As String, ByVal maxLength As Integer) As String
        If String.IsNullOrEmpty(value) Then Return value
        If value.Length > maxLength Then Console.WriteLine(String.Format("Truncate performed: {0} to {1}", value, value.Substring(0, maxLength)))
        Return If(value.Length <= maxLength, value, value.Substring(0, maxLength))
    End Function
    <Extension()>
    Public Function CenterTo(ByVal value As String, ByVal lenght As Integer) As String
        value = value.Trim()
        If String.IsNullOrEmpty(value) Then Return value
        If value.Length > lenght Then Return value
        Return value.PadLeft(value.Length + ((lenght - value.Length) / 2), " "c)
    End Function
End Module
